### DISCUSSION TITLE
Where can i create main logic class ?

### DISCUSSION THREAD

**Discussion by ilia19911** [AvaloniaUI/Avalonia#6241] [Q&A]
hi. I think, my program can be both console and gui. Usually, if i use mvvm pattern in XAML, then I start my program main logic in Program class as static. I guess that it wrong, and if start it in avalonia, then you can find in Program class it
        // Initialization code. Don't use any Avalonia, third-party APIs or any
        // SynchronizationContext-reliant code before AppMain is called: things aren't initialized
        // yet and stuff might break.
Where can i stay my program logic in avalonia ?

**Comment by maxkatz6** [comment]
Usually, it's in App.xaml Initialize method.
You can also use ReactiveUI WhenActivated in the specific windows and user controls.
